### PR TITLE
All changes happen in single transaction

### PR DESCRIPTION
**Pull request by forus**

### CODE REVIEW

File: transmart-copy/README.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ _Parameters:_
 - `-w <file>`, `--write <file>`: Write observations to TSV file `<file>`.
 - `-v`, `--vacuum-analyze`: Vacuum analyze the `observation_fact` table.
 - `-d`, `--directory`: Specifies a data directory.
+- `-ru`, `--re-upload`: Removes and loads the studies specified in the data directory.
```
Comment (gijskant) [inline transmart-copy/README.md]: But why not reuse the `-d` parameter then?

File: transmart-copy/src/test/groovy/org/transmartproject/copy/CopySpec.groovy [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
-    static STUDY_FOLDER = './src/main/resources/examples/' + TEST_STUDY
-    static DATABASE_CREDENTIALS = [
-            PGUSER: 'i2b2demodata',
+    static String STUDY_FOLDER = './src/main/resources/examples/' + TEST_STUDY
+    static String CORRUPTED_STUDY_FOLDER = './src/main/resources/examples/' + TEST_STUDY + '_corrupted'
+    static Map<String, String> DATABASE_CREDENTIALS = [
+            PGUSER    : 'i2b2demodata',
             PGPASSWORD: 'i2b2demodata'
-    ]
+    ].withDefault { String key -> System.getenv(key) }
```
Comment (gijskant) [inline transmart-copy/src/test/groovy/org/transmartproject/copy/CopySpec.groovy]: Is there a reason to include the tester's environment variables in the test?

File: transmart-copy/src/main/groovy/org/transmartproject/copy/Copy.groovy [hunk 3]
```diff
@@ -140,61 +127,90 @@ class Copy {
...
                 def studyId = cl.getOptionValue('delete')
-                try {
-                    copy.deleteStudy(studyId)
-                } catch(Exception e) {
-                    log.error "Error deleting study ${studyId}."
-                    throw e
-                }
+                copy.deleteStudyById(studyId)
+            }
+            if (cl.hasOption('re-upload')) {
```
Comment (gijskant) [inline transmart-copy/src/main/groovy/org/transmartproject/copy/Copy.groovy]: Twice checking for re-upload...

File: transmart-copy/src/main/groovy/org/transmartproject/copy/Copy.groovy [hunk 4]
```diff
@@ -140,61 +127,90 @@ class Copy {
...
+                def directory = cl.getOptionValue('re-upload')
+                copy.deleteStudy(directory, false)
             }
             if (cl.hasOption('drop-indexes')) {
                 copy.dropIndexes()
             }
-            if (cl.hasOption('mode')) {
+            boolean independentOperationOptionSpecified = INDEPENDENT_OPERATION_OPTIONS
```
Comment (gijskant) [inline transmart-copy/src/main/groovy/org/transmartproject/copy/Copy.groovy]: The control flow is really unclear here.

File: transmart-copy/README.md (hunk 1 above)
Comment (forus) [reply transmart-copy/README.md]: ah, true. So rewriting a study would be the default behavior.

File: transmart-copy/src/main/groovy/org/transmartproject/copy/Copy.groovy (hunk 4 above)
Comment (forus) [reply transmart-copy/src/main/groovy/org/transmartproject/copy/Copy.groovy]: It became a little bit simpler by removing `--re-upload` option ;)

**Pull request merged**